### PR TITLE
Issue #74: PyCSW update and delete

### PR DESCRIPTION
**Pull request by vkbo**
**Summary**:

**Related issue**:

**Suggested reviewer(s)**:

**Reviewer checklist**:

* [ ] The headers of all files contain a reference to the repository license
* [ ] 100% test coverage of new code - meaning:
  * [ ] The overall test coverage increased or remained the same as before
  * [ ] Every function is accompanied with a test suite
  * [ ] Tests are both positive (testing that the function work as intended with valid data) and negative (testing that the function behaves as expected with invalid data, e.g., that correct exceptions are thrown)
  * [ ] Functions with optional arguments have separate tests for all options
* [ ] Examples are supported by doctests
* [ ] All tests are passing
* [ ] All names (e.g., files, classes, functions, variables) are explicit
* [ ] Documentation (as docstrings) is complete and understandable

The checklist is based on the S-ENDA conventions and definition of done (see [General Conventions](https://s-enda-documentation.readthedocs.io/en/latest/general_conventions.html)). The above points are not necessarily relevant to all contributions. In that case, please add a short explanation to help the reviewer.

### CODE REVIEW

**Comment by vkbo** [comment]
All the bits and pieces are in place, but we really need some integration tests now to check that the full chain works.

We can do that in a separate PR, or in this one, but I wouldn't bump the version until we've added such tests.

File: dmci/distributors/pycsw_dist.py [hunk 1]
```diff
@@ -119,16 +147,16 @@ def _delete(self):
...
-            '       <csw:Constraint version="1.1.0">'
-            '           <ogc:Filter>'
-            '               <ogc:PropertyIsEqualTo>'
-            '                   <ogc:PropertyName>apiso:Identifier</ogc:PropertyName>'
-            '                   <ogc:Literal>%s</ogc:Literal>'
-            '               </ogc:PropertyIsEqualTo>'
-            '           </ogc:Filter>'
-            '       </csw:Constraint>'
-            '   </csw:Delete>'
+            '  <csw:Delete>'
```
Comment (mortenwh) [inline dmci/distributors/pycsw_dist.py:150]: should we also define this as binary (i.e., like `b'some text'`)?

Comment (johtoblan) [reply dmci/distributors/pycsw_dist.py:150]: requests.post handles both str and bytes, so I think we can keep it like this (To not have to encode _metadata_id, as it is str)

**Review by johtoblan — APPROVED** [review]
LGTM

**Pull request merged**